### PR TITLE
prov/efa,fabtests/pytest: fix a bug in pytest that hide failure

### PR DESCRIPTION
**Pull request by wzamazon**
This PR contains 2 patches:

1st one fixed a bug in pytest script that hide a testing failure.
2nd one fixed a bug in EFA provider that can cause the failure.

### CODE REVIEW

**Comment by wzamazon** [comment]
AWS CI is expected to fail, because is because I have not pushed the 2nd commit, and the 1st commit fixed the bug in testing script that will hide a failure.

**Comment by wzamazon** [comment]
AWS CI failed as expected.

Now I will add the 2nd commit, which should fix the failure.

**Comment by wzamazon** [comment]
2nd commit added. Ready for review.

File: prov/efa/src/efa_cq.c [hunk 1]
```diff
@@ -160,12 +160,12 @@ ssize_t efa_cq_readfrom(struct fid_cq *cq_fid, void *buf, size_t count,
 		}
 
 		if (src_addr) {
-			av = cq->domain->qp_table[wc.ibv_wc.qp_num &
-				cq->domain->qp_table_sz_m1]->ep->av;
+			qp_num = ibv_wc_read_qp_num(cq->ibv_cq_ex);
+			src_qp = ibv_wc_read_src_qp(cq->ibv_cq_ex);
+			slid = ibv_wc_read_slid(cq->ibv_cq_ex);
+			av = cq->domain->qp_table[qp_num & cq->domain->qp_table_sz_m1]->ep->av;
```
Comment (shijin-aws) [inline prov/efa/src/efa_cq.c:166]: so, before the fix, qp_num, src_qp, and slid was actually zero? Which test is failing due to this bug?

Comment (wzamazon) [reply prov/efa/src/efa_cq.c:166]: Yes. all of them were 0.

The `fi_dgram_pingpong` test was failing because of this bug.

Comment (wzamazon) [reply prov/efa/src/efa_cq.c:166]: But `runfabtests.py` did not report the failure due to the bug in `common.py`

Comment (shijin-aws) [reply prov/efa/src/efa_cq.c:166]: could you explain why it only impacts dgram?

Comment (wzamazon) [reply prov/efa/src/efa_cq.c:166]: only dgram endpoint call the function `efa_cq_readfrom`.

rdm endpoint call `ofi_cq_read` which call `rxr_ep_progress` then call `ibv_start_poll` and `ibv_end_poll` directly, bypassing `efa_cq_readfrom`.

BTW, we should rename the file `efa_cq.c` to `efa_dgram_cq.c` and all functions in this file `efa_dgram_xxx`.

**Comment by wzamazon** [comment]
AWS CI passed as expected.

**Review by shijin-aws — APPROVED** [review]

**Pull request merged**